### ISSUE TITLE
Add warning for $command && successful || unsuccessful

### ISSUE THREAD

**Issue by brianjmurrell** [koalaman/shellcheck#2531]
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
it's hard to search as GH's search probably doesn't consider && and || in the search terms as something to search for

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
true && { echo "true"; false; } || echo "false"
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Don't use && ... || in place of if...then...else...fi as they do not operate equally in all cases.

Here is the case where the above syntax is troublesome:

```sh
$ true && { echo "true"; false; } || echo "false"
true
false
```

Which is not likely what the writer expected.  I suspect they expect the same handling as:

```sh
$ if true; then echo "true"; false; else echo "false"; fi
true
```

**Comment by koalaman** [comment]
There's already a warning for `A && B || C` (https://www.shellcheck.net/wiki/SC2015), it just happens to be suppressed when C is `echo/printf/exit/return` as these tend to be benign.

If you change it to e.g. `true && { echo "true"; false; } || rm -rf foo` you'll see `SC2015 (info): Note that A && B || C is not if-then-else. C may run when A is true.`
